api/src/test/java/uk/gov/dhsc/htbhf/claimant/reporting/GoogleAnalyticsClientTest.java

```diff
@@ -0,0 +1,88 @@
+import uk.gov.dhsc.htbhf.claimant.exception.GoogleAnalyticsException;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowableOfType;
+import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.RANDOM_PORT;
+import static org.springframework.http.MediaType.TEXT_PLAIN_VALUE;
+
+@SpringBootTest(webEnvironment = RANDOM_PORT)
+@AutoConfigureEmbeddedDatabase
+@AutoConfigureWireMock(port = 8120)
+class GoogleAnalyticsClientTest {
+
+    private static final String REPORT_ENDPOINT = "/collect";
+    // properties need to be ordered for the assertion against the encoded string to work
+    private static final Map<String, String> REPORT_PROPERTIES = new LinkedHashMap<>();
+    // google analytics expects the payload to be url encoded and in the format of a query string
+    private static final String URL_ENCODED_REPORT_PROPERTIES = "cd8=Bristol+West&cd9=NHS+Bristol%2C+North+Somerset+and+South+Gloucestershire";
+
+    static {
+        REPORT_PROPERTIES.put("cd8", "Bristol West");
+        REPORT_PROPERTIES.put("cd9", "NHS Bristol, North Somerset and South Gloucestershire");
+    }
+
+    @Value("${google-analytics.base-uri}")
+    private String baseUri;
+
+    @Autowired
+    private GoogleAnalyticsClient googleAnalyticsClient;
+
+    @AfterEach
+    @SuppressWarnings("PMD.UnnecessaryFullyQualifiedName")
+    void tearDown() {
+        WireMock.reset();
+    }
+
+    @Test
+    void shouldCallGoogleAnalyticsWithUrlEncodedQueryStringPayload() {
+        stubFor(post(urlEqualTo(REPORT_ENDPOINT)).withHeader("Content-Type", equalTo(TEXT_PLAIN_VALUE))
+                .willReturn(ok()));
+
+        googleAnalyticsClient.reportEvent(REPORT_PROPERTIES);
+
+        verify(exactly(1), postRequestedFor(urlEqualTo(REPORT_ENDPOINT))
+                .withHeader("Content-Type", equalTo(TEXT_PLAIN_VALUE))
+                .withRequestBody(equalTo(URL_ENCODED_REPORT_PROPERTIES)));
+    }
+
+    @Test
+    void shouldThrowExceptionWhenGoogleAnalyticsReturnsANonOkResponseCode() {
+        stubFor(post(urlEqualTo(REPORT_ENDPOINT)).withHeader("Content-Type", equalTo(TEXT_PLAIN_VALUE))
+                .willReturn((noContent())));
+
+        GoogleAnalyticsException exception = catchThrowableOfType(() -> googleAnalyticsClient.reportEvent(REPORT_PROPERTIES), GoogleAnalyticsException.class);
+
+        assertThat(exception.getMessage()).isEqualTo("Response code from google analytics was not as expected, received: NO_CONTENT");
+        verify(exactly(1), postRequestedFor(urlEqualTo(REPORT_ENDPOINT))
+                .withHeader("Content-Type", equalTo(TEXT_PLAIN_VALUE))
+                .withRequestBody(equalTo(URL_ENCODED_REPORT_PROPERTIES)));
+    }
+
+    @Test
+    void shouldThrowExceptionWhenGoogleAnalyticsReturnsAndErrorResponse() {
```

Comment:
typo: should be `shouldThrowExceptionWhenGoogleAnalyticsReturnsAnErrorResponse` (currently has `...AndErrorResponse`)